Rewrite/src/Rewrite.CSharp/CSharpPrinter.cs

```diff
@@ -94,6 +94,22 @@ public override Cs VisitCompilationUnit(Cs.CompilationUnit compilationUnit, Prin
         return compilationUnit;
     }
 
+    public override J? VisitClassDeclaration(Cs.ClassDeclaration classDeclaration, PrintOutputCapture<TState> p)
+    {
+        _delegate.VisitClassDeclaration(classDeclaration.ClassDeclarationCore, p);
+        return classDeclaration;
+    }
+
+    public override J? VisitMethodDeclaration(Cs.MethodDeclaration methodDeclaration, PrintOutputCapture<TState> p)
+    {
+        base.VisitMethodDeclaration(methodDeclaration, p);
+        if (methodDeclaration.MethodDeclarationCore.Body == null)
+        {
+            p.Append(";");
```

Comment:
Here I wonder what about trivia before the semicolon. Isn't this element wrapped inside a JRightPadded to store that?

Reply:
You're right, it passes in cases where there are generics in use (because constraints are in JContainer, so trailing whitespace is properly captured), but in other instances they are not:
```
E| void Method()  /*1*/ ;␍␤
A| void Method(); /*1*/  ␍␤
```
How do you suggest we model it? I can't wrap the `J.ClassDefinition` in `JRightPadded` because generics stuff goes "after it", and if I wrap `Cs.ClassDefinition` which level would be responsible for processing semicolon?

I guess I can force the use of `Block` which already has semantics to render itself as semicolon if it has a marker.

Reply:
Isn't the type parameter handing missing here anyway? I must have missed that here on my phone.

Using the body for that would be one option, as it would then be the body's prefix.